### DISCUSSION TITLE
Why did i get the best benchmark if i set the max connection to 1 ?

### DISCUSSION THREAD

**Discussion by masfahru** [porsager/postgres#648] [Q&A]
I have tested some popular database drivers (postgres, pg, mariadb, mysql2) using TechEmpower benchmark tools, i use 3 VMs (server, database, and client, each has 8 vcpus / 8 GB memory)

Tested using the **default options** of each database driver, this is the result:
![image](https://github.com/porsager/postgres/assets/92277627/aeaa2e5c-9cf7-41b6-8466-2ae0b10d2855)
https://www.techempower.com/benchmarks/#section=test&shareid=b3cef5dd-6bac-420c-996f-26e3a72d97eb&test=db

Changed the `max` connections for postgres to `max: 1`, and this is the result:
![image](https://github.com/porsager/postgres/assets/92277627/eeb16dc7-4396-4832-91ce-65343024e6d8)
https://www.techempower.com/benchmarks/#section=test&shareid=c6a707de-3738-4c59-9501-55403c99b35d&test=db

Why did i get the best benchmark if i set the `max` connection to 1 ?

**Comment by porsager** [comment]
Cool.. Can you share the code / repo, then I can take a look?

**Reply by Dhruv-Garg79** [reply]
hey did you find out anything? I am also interested to know.

**Reply by luoxzhg** [reply]
[an example in techEmpower fastify bench](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/JavaScript/fastify/db/postgres.js)
